openquake/hazardlib/gsim/kuehn_2020.py

```diff
@@ -0,0 +1,761 @@
+                pga_soil = get_mean_values(C_PGA, self.region, trt, m_b,
+                                           ctx, ctx, ctx, pga1100)
+                break
+
+        for m, imt in enumerate(imts):
+            # Get coefficinets for imt
+            C = self.COEFFS[imt]
+            m_break = m_b + C["dm_b"] if trt == const.TRT.SUBDUCTION_INTERFACE\
+                else m_b
+
+            if imt.string == "PGA":
+                mean[m] = pga_soil
+            elif ("SA" in imt.string) and (imt.period <= 0.1):
+                # If Sa (T) < PGA for T <= 0.1 then set mean Sa(T) to mean PGA
+                mean[m] = get_mean_values(C, self.region, trt, m_break,
+                                          ctx, ctx, ctx, pga1100)
+                idx = mean[m] < pga_soil
+                mean[m][idx] = pga_soil[idx]
+            else:
+                # For PGV and Sa (T > 0.1 s)
+                mean[m] = get_mean_values(C, self.region, trt, m_break,
+                                          ctx, ctx, ctx, pga1100)
+            # Apply the sigma mu adjustment if necessary
+            if self.sigma_mu_epsilon:
+                sigma_mu_adjust = get_sigma_mu_adjustment(self.sigma_mu_model,
+                                                          imt, ctx, ctx)
+                mean[m] += (self.sigma_mu_epsilon * sigma_mu_adjust)
+            # Get standard deviations
+            tau[m] = C["tau"]
+            phi[m] = C["phi"]
+        sig[:] = np.sqrt(tau ** 2.0 + phi ** 2.0)
+
+    # Coefficients in external file - supplied directly by the author
+    COEFFS = CoeffsTable(sa_damping=5, table=open(KUEHN_COEFFS).read())
+
+
+class KuehnEtAl2020SSlab(KuehnEtAl2020SInter):
+    """
+    Implements NGA Subduction model of Kuehn et al. (2020) for Intraslab events
+
+    This class implements the global model.
+    Adjustments with respect to the interface model are
+        different constant
+        different magnitude scaling coefficent
+        different geometrical spreading coefficient
+        no magnitude break adjustment at long periods
+        different depth scaling and adjustment to break point
+        different depth centering and break point
+        different default magnitude break point
+    """
+
+    #: Supported tectonic region type is subduction in-slab
+    DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.SUBDUCTION_INTRASLAB
+
+
+# For the aliases use the verbose form of the region name
+REGION_ALIASES = {
+    "GLO": "",
+    "USA-AL": "Alaska",
+    "CAS": "Cascadia",
+    "CAM": "CentralAmericaMexico",
+    "JPN": "Japan",
+    "NZL": "NewZealand",
+    "SAM": "SouthAmerica",
+    "TWN": "Taiwan",
+}
+
+
+for region in SUPPORTED_REGIONS[1:]:
+    add_alias("KuehnEtAl2021SInter" + REGION_ALIASES[region],
+              KuehnEtAl2020SInter,
+              region=region)
+    add_alias("KuehnEtAl2021SSlab" + REGION_ALIASES[region],
+              KuehnEtAl2020SSlab,
+              region=region)
```

Comment:
Good work in using the new alias feature!

Reply:
@g-weatherill & @micheles should the alias be `KuehnEtAl2020SSlab + ...` instead of `KuehnEtAl2021SSlab + ...`?